dbt_platform_helper/domain/maintenance_page.py

```diff
@@ -262,27 +265,48 @@ def add_maintenance_page(
                 ],
             )
         except Exception as e:
-            deleted_rules = self.__clean_up_maintenance_page_rules(listener_arn)
+            self.__clean_up_maintenance_page_rules(listener_arn)
             raise FailedToActivateMaintenancePageException(
-                app, env, f"{e}:\n {traceback.format_exc()}", deleted_rules
+                app, env, f"{e}:\n {traceback.format_exc()}"
             )
 
     def __clean_up_maintenance_page_rules(
         self, listener_arn: str, fail_when_not_deleted: bool = False
-    ) -> dict[str, bool]:
+    ):
 
         tag_descriptions = self.load_balancer.get_rules_tag_descriptions_by_listener_arn(
             listener_arn
         )
 
-        deletes = {}
+        # keep track of rules deleted
+        deletes = {"MaintenancePage": 0}
```

Comment:
Could it be named such that the comment is not needed?
```suggestion
        deleted_rules = {"MaintenancePage": 0}
```